UnitTests/UnitTests.XamlIslands/XamlIslandsTest_ThemeListener_Threading.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Uwp.UI.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Windows.UI.Xaml;
+
+namespace UnitTests.XamlIslands
+{
+    [STATestClass]
+    public partial class XamlIslandsTest_ThemeListener_Threading
+    {
+        private TaskCompletionSource<object> _taskCompletionSource;
+        private ThemeListener _themeListener = null;
+
+        [TestInitialize]
+        public void Init()
+        {
+            Application.Current.RequestedTheme = ApplicationTheme.Light;
+
+            _taskCompletionSource = new TaskCompletionSource<object>();
+
+            _themeListener = new ThemeListener();
+            _themeListener.ThemeChanged += (s) =>
+            {
+                _taskCompletionSource.SetResult(null);
```

Comment:
>SetResult [](start = 38, length = 9)

If ThemeChanged fires multiple times, this will throw - you can use TrySetResult to not throw on the second attempt, if that matters to you.

Reply:
It should fire only once. I think its better to be specific here.